eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/ErrorHandlers.scala

```diff
@@ -132,13 +132,20 @@ trait ErrorHandlers extends CommonHandlers {
       case negotiating@DATA_NEGOTIATING(_, _, _, _, Some(bestUnpublishedClosingTx)) =>
         // if we were in the process of closing and already received a closing sig from the counterparty, it's always better to use that
         handleMutualClose(bestUnpublishedClosingTx, Left(negotiating))
+      // The channel was never used and the funding tx could be double-spent: we don't need to publish our commitment
+      // since we don't have funds in the channel, but it's a nice thing to do because it lets our peer get their
+      // funds back without delays if they can't double-spend the funding tx.
       case d: DATA_WAIT_FOR_FUNDING_CONFIRMED if Closing.nothingAtStake(d) =>
-        // The channel was never used and the funding tx could be double-spent: we don't need to publish our commitment
-        // since we don't have funds in the channel, but it's a nice thing to do because it lets our peer get their
-        // funds back without delays if they can't double-spend the funding tx.
         val commitTx = d.commitments.fullySignedLocalCommitTx(keyManager)
         txPublisher ! PublishFinalTx(commitTx, 0 sat, None)
         goto(CLOSED)
+      case d: DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED if Closing.nothingAtStake(d) =>
+        val commitTx = d.commitments.fullySignedLocalCommitTx(keyManager)
+        txPublisher ! PublishFinalTx(commitTx, 0 sat, None)
+        goto(CLOSED)
+      case _: DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED =>
+        log.info("cannot close channel while dual-funding txs are unconfirmed: waiting for a transaction to confirm or be double-spent")
```

Comment:
I propose a different approach:
- we add `alternativeCommitments: Seq[Commitments]` to `DATA_CLOSING`, which will be non-empty if we transition from `WAIT_FOR_FUNDING_CONFIRMED`->`CLOSING`
- in `CLOSING`, there are two cases:
  - either the most recent funding tx confirms: nominal case, nothing to do
  - an older funding tx confirms: we build a fresh `CLOSING` for the commitments that confirmed (we can forget all other commitments, and all child transactions, they have all been double-spent).

Reply:
Good idea, done in https://github.com/ACINQ/eclair/pull/2274/commits/075a606665582c54862b63202d2c0fd68b3c4b1a